dev/devicelab/bin/tasks/flutter_test_performance.dart

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:path/path.dart' as path;
+
+import 'package:flutter_devicelab/framework/framework.dart';
+import 'package:flutter_devicelab/framework/utils.dart';
+
+final RegExp testOutputPattern = new RegExp(r'^[0-9][0-9]:[0-9][0-9] \+[0-9]+: (.+?) *$');
+
+Future<int> runTest() async {
+  final Stopwatch clock = new Stopwatch()..start();
+  final Process analysis = await startProcess(
+    path.join(flutterDirectory.path, 'bin', 'flutter'),
+    <String>['test', path.join('flutter_test', 'trivial_widget_test.dart')],
+    workingDirectory: path.join(flutterDirectory.path, 'dev', 'automated_tests'),
+  );
+  int badLines = 0;
+  int step = 0;
+  await for (String entry in analysis.stdout.transform(utf8.decoder).transform(const LineSplitter())) {
+    print('test stdout ($step): $entry');
+    if (step == 0 && entry == 'Building flutter tool...') {
+      // ignore this line
+      step = 1;
+    } else if (step <= 1 && entry == 'Running "flutter packages get" in automated_tests...') {
+      // ignore this line
+      step = 2;
+    } else if (step < 3 && entry == '') {
+      // we have a blank line at the start
+      step = 3;
+    } else {
+      final Match match = testOutputPattern.matchAsPrefix(entry);
+      if (match == null) {
+        badLines += 1;
+      } else {
+        if (step >= 3 && step <= 4 && match.group(1).startsWith('loading ')) {
+          // first the test loads
+          step = 4;
+        } else if (step <= 5 && match.group(1) == 'A trivial widget test') {
+          // then the test runs
+          step = 5;
+        } else if (step < 6 && match.group(1) == 'All tests passed!') {
+          // then the test finishes
+          step = 6;
+        } else {
+          badLines += 1;
+        }
+      }
+    }
+  }
+  await for (String entry in analysis.stderr.transform(utf8.decoder).transform(const LineSplitter())) {
+    print('test stderr: $entry');
+    badLines += 1;
+  }
+  final int result = await analysis.exitCode;
+  clock.stop();
+  if (result != 0)
+    throw new Exception('flutter test failed with exit code $result');
+  if (badLines > 0)
+    throw new Exception('flutter test renderered unexpected output ($badLines bad lines)');
+  if (step != 6)
+    throw new Exception('flutter test did not finish (only reached step $step)');
```

Comment:
nit: can we make the message say the step name instead of number (one way to do it would be to use an enum for step)

Reply:
good call on the enum. done.